replication-api-impl/src/main/java/com/connexta/replication/api/impl/worker/Worker.java

```diff
@@ -0,0 +1,354 @@
+
+      final OperationType type = task.getOperation();
+      if (type == OperationType.HARVEST) {
+        doCreate(task, metadata, remote, local);
+      } else {
+        // should never hit this since deserialization will return UNKNOWN
+        task.unlock();
+      }
+    } finally {
+      try {
+        remote.close();
+      } catch (IOException e) {
+        LOGGER.debug("Failed to close destination adapter", e);
+      }
+    }
+  }
+
+  @Nullable
+  private NodeAdapter getDestinationAdapter(Task task) {
+    Site site;
+    final String siteId = task.getQueue().getSite();
+    try {
+      site = siteManager.get(siteId);
+    } catch (NotFoundException e) {
+      LOGGER.debug("Unable to find site {} for queue", siteId, e);
+      return null;
+    } catch (ReplicationPersistenceException e) {
+      LOGGER.debug("Failed to fetch site {} for queue", siteId, e);
+      return null;
+    }
+
+    NodeAdapter nodeAdapter;
+    try {
+      nodeAdapter = nodeAdapters.factoryFor(site.getType()).create(site.getUrl());
+    } catch (IllegalArgumentException | AdapterException e) {
+      LOGGER.debug("Error retrieving node adapter factory for site {}", site.getUrl(), e);
+      return null;
+    }
+
+    return nodeAdapter;
+  }
+
+  private void doCreate(
+      final Task task,
+      final Metadata metadata,
+      final NodeAdapter source,
+      final NodeAdapter destination)
+      throws InterruptedException {
+    final String metadataId = task.getId();
+    final String operation = task.getOperation().name();
+
+    try {
+      // TODO: remove existence check once better reporting is done by adapters
+      // see https://github.com/connexta/replication/issues/235
+      if (!source.exists(metadata)) {
+        LOGGER.info(
+            "Received task to create metadata on site {}, but it was deleted on {}",
+            destination.getSystemName(),
+            source.getSystemName());
+        task.fail(
+            ErrorCode.NO_LONGER_EXISTS,
+            String.format(
+                "Metadata deleted from site %s before able to create", source.getSystemName()));
+        return;
+      }
+    } catch (AdapterInterruptedException e) {
+      throw new InterruptedException(e.getMessage());
+    } catch (AdapterException e) {
+      LOGGER.debug("Failed to check if resource exists", e);
+      task.fail(
+          ErrorCode.OPERATION_FAILURE,
+          String.format("Failed check resource existence on %s", destination.getSystemName()));
+      return;
+    }
+
+    URI uri = task.getResource().flatMap(ResourceInfo::getResourceUri).orElse(null);
+    if (uri != null) {
+      ResourceRequest request = new ResourceRequestImpl(metadata);
+      ResourceResponse response;
+      try {
+        response = source.readResource(request);
+      } catch (AdapterInterruptedException e) {
+        throw new InterruptedException(e.getMessage());
```

Comment:
✏️ Don't bother with the message since it won't ever have one. Either create it blank or pass the exception as a cause only.